src/native/libs/System.Security.Cryptography.Native.Apple/CMakeLists.txt

```diff
@@ -38,6 +39,31 @@ else()
     )
 endif()
 
+# As of CMake 3.20.2 support for Swift only works with the Ninja and XCode
+# generators so we cannot rely on it. Even with the Ninja generator it doesn't
+# work in combination with other languages within the same library.
+if (NOT SWIFT_COMPILER_TARGET AND CLR_CMAKE_TARGET_OSX)
```

Comment:
Can we instead use `enable_language(Swift)` and then take advantage of `CMAKE_${Lang}_XYZ` auto configurations?

* https://github.com/dotnet/runtime/blob/6583aa1c3f65da8879d188411d07d80108051269/eng/native/configurecompiler.cmake#L798
* https://github.com/Kitware/CMake/blob/9cd719b8f368df4f1636c9b79adea93d68008a46/Tests/RunCMake/XcodeProject/XCTestAddBundle.cmake#L1

Reply:
I don't think so, the comment above tries to address why we can't, unless I am misunderstanding your suggestion.

Reply:
OK, sounds like you have tried `enable_language(Swift)` and it didn't work outside XCode. Browsing through the code in `Modules/Platform/Apple-Apple-Swift.cmake` etc. I don't see what is preventing it.

Reply:
I end up hitting https://github.com/Kitware/CMake/blob/381fc368807d67dba58db8253ab937c082f8f3a3/Modules/CMakeDetermineSwiftCompiler.cmake#L45